wpimath/src/main/native/include/frc/geometry/Pose2d.h

```diff
@@ -233,3 +263,71 @@ void from_json(const wpi::json& json, Pose2d& pose);
 #include "frc/geometry/proto/Pose2dProto.h"
 #endif
 #include "frc/geometry/struct/Pose2dStruct.h"
+
+#include "frc/geometry/Transform2d.h"
+
+namespace frc {
+
+constexpr Transform2d Pose2d::operator-(const Pose2d& other) const {
+  const auto pose = this->RelativeTo(other);
+  return Transform2d{pose.Translation(), pose.Rotation()};
+}
```

Comment:
Why are these out-of-line instead of in the class body?

Reply:
Circular dependency between Pose2d and Transform2d.